demosplan/DemosPlanCoreBundle/Command/Documentation/EventFinder.php

```diff
@@ -0,0 +1,340 @@
+
+    /**
+     * The list of events, created by this command may be incomplete and/or contains false-positives,
+     * caused by limited options to identify relevant events as such, while event classes are not loaded.
+     * To do so, the name of the class will be used too, by looking for the term "event" as part of the name.
+     */
+    protected function configure(): void
+    {
+        $this->addOption(
+            self::OPTION_PARENTS,
+            'p',
+            InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+            'The parent class name(s) which will be used as filter to determine the correct event-classes.'
+        );
+
+        //-s path/to/dir/one -s path/to/dir/two
+        $this->addOption(
+            self::OPTION_START_PATHS,
+            's',
+            InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+            'Start path(s), where to search for event classes.'
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        try {
+            $startPaths = $input->getOption(self::OPTION_START_PATHS);
+            $targetParentClassNames = $input->getOption(self::OPTION_PARENTS);
+            //add root-path of demosplan will be searched anyway
+            $startPaths[] = DemosPlanPath::getRootPath().'demosplan';
+
+            $phpFilePaths = self::findPhpClassFilesInDirectories($startPaths);
+            $this->collectEventMatches($phpFilePaths, $targetParentClassNames);
+            $this->findUsagesOfEvents($phpFilePaths, $this->namedEventMatches);
+
+            //Convert content to array to allow encode to json
+            $this->namedEventMatches = array_map(
+                static fn(EventMatch $eventMatch): array => $eventMatch->toArray(),
+                $this->namedEventMatches
+            );
+
+            //Convert content to array to allow encode to json
+            $this->unnamedEventMatches = array_map(
+                static fn(array $nestedEventMatches): array => array_map(
+                    static fn(UnnamedEventMatch $eventMatch): array => $eventMatch->toArray(),
+                    $nestedEventMatches
+                ),
+                $this->unnamedEventMatches
+            );
+
+            $output->writeln(
+                Json::encode([
+                    'named Events' => $this->namedEventMatches,
+                    'unnamed Events' => $this->unnamedEventMatches,
+                ],
+                    \JSON_PRETTY_PRINT)
+            );
+
+        } catch ( JsonException ) {
+            $output->writeln('{"error": "Event export failed."}');
+
+            return Command::FAILURE;
+        }
+
+        return Command::SUCCESS;
+    }
+
+
+    /**
+     * Collects all php files within the given directory, recursively.
+     *
+     * @return list<string>
+     */
+    private static function findPhpClassFilesInDirectory(string $directory): array
```

Comment:
Why static?

Reply:
PhpStorm notice me, that this method can be static. I don't see why this should not be static, that's why I followed the suggestion and made the method static.
Do you see any disadvantages?